docs/src/reference/builtins.md

```diff
@@ -0,0 +1,317 @@
+
+**Signature:** `([a: string] [b: string] [c: record] -- record)`
+
+**Equivalent Rust:** `c.insert(b, a)`
+
+### Property (`prop`)
+
+**Signature:** `([a: string] [b: record] -- any)`
+
+**Equivalent Rust:** `b.get(a)`
+
+### Has (`has`)
+
+**Signature:** `([a: record] [b: string] -- bool)`
+
+**Equivalent Rust:** `a.has(b)`
+
+### Remove (`remove`)
+
+**Signature:** `([a: record] [b: string] --)`
+
+**Equivalent Rust:** `a.remove(b)`
+
+### Keys (`keys`)
+
+**Signature:** `([a: record] -- list(symbol))`
+
+**Equivalent Rust:** `a.keys()`
+
+### Values (`values`)
+
+**Signature:** `([a: record] -- list(any))`
+
+**Equivalent Rust:** `a.values()`
+
+## Types
+
+### Cast (`cast`)
+
+**Signature:** `([a] [b: string] -- any)`
+
+Converts `a` to the type: `b` and returns the new type
+
+### Type of (`typeof`)
+
+**Signature:** `([a] -- string)`
+
+Gets the type of `a` and pushes it as a string to the stack
+
+### Lazy (`lazy`)
+
+**Signature:** `([a] -- lazy(a))`
+
+Wraps `a` with a lazy expression, making it lazy.
+
+## Control Flow
+
+### If (`if`)
+
+**Signature:** `([a: list] [b: bool] --)`
+
+**Equivalent Rust:** `if b { a }`
+
+### Halt (`halt`)
+
+**Signature:** `(--)`
+
+**Equivalent Rust:** Halts execution.
+
+### Recur (`recur`)
+
+**Signature:** `(-- symbol)`
+
+A QoL helper intrinsic that pushes the symbol: `recur` to the stack. Used to allow `recur` to be called without escaping with a lazy (such as `'recur`).
+
+### OrElse ('orelse')
+
+**Signature:** `([a] [b] -- a|b)`
+
+**Equivalent Rust:** `a.or(b)`
+
+If `a` is `nil`, returns `b`. Else, returns `a`.
+
+## Scopes and Variables
+
+### Call (`call`)
+
+**Signature:** `([a] --)`
+
+Calls `a` and:
+- If `a` is a **function**: Runs the function
+- If `a` is a **list**: Runs each item in the list
+- If `a` is a **symbol**: Calls the symbol from the scope
+- If `a` is **anything else**: Pushes it back onto the stack
+
+### Let (`let`)
+
+**Signature:** `([a: list] [b: list(symbol)] --)`
+
+Pops `b.len()` items off of the stack, assigning each item the corresponding symbol in `b`. Then, runs the code block `a`, injecting the symbols into the scope.
+
+If list `b` was `(first second)`, then they would be popped from the stack in order, following this signature: `([first] [second] --)`.
+
+### Define (`def`)
+
+**Signature:** `([a] [b: symbol] --)`
+
+**Equivalent Rust:** `let b = a`
+
+### Set (`set`)
+
+**Signature:** `([a] [b: symbol] --)`
+
+**Equivalent Rust:** `b = a`
+
+### Get (`get`)
+
+**Signature:** `([a: symbol] -- any)`
+
+**Equivalent Rust:** `a`
+
+## Debugging and I/O
+
+### Debug (`debug`)
+
+**Signature:** `([a] -- a)`
+
+**Equivalent Rust:** `dbg!(a)`
+
+### Assert (`assert`)
+
+**Signature:** `([a] [b] -- a)`
+
+**Equivalent Rust:** `assert!(b, format!("{}", a))`
+
+### Print (`print`)
+
+**Signature:** `([a] --)`
+
+**Equivalent Rust:** `println!("{}", a)`
+
+### Pretty (`pretty`)
+
+**Signature:** `([a] --)`
+
+**Equivalent Rust:** `println!("{:#}", val)`
```

Comment:
Is it worth mentioning these? Given that they will be part of the standard-library soon-ish.